### ISSUE TITLE
 About CEL Loss

### ISSUE THREAD

**Issue by LYJ903118120** [lartpang/MINet#16]
Hello, may CEL loss be Dice loss, right?

**Comment by lartpang** [comment]
Thank you for your attention. Actually, it's a bit of a coincidence. In terms of the final form, cel can indeed be transformed into a form similar to dice loss. 

But the difference is that when we build this loss, we derive it **intuitively according to the needs of the task itself and the proportional relationship between regions.** 
- The form of the area ratio helps to mitigate the fluctuation of loss caused by the size of the object.
- We **only pay attention to the difference region**, that is FP+FN, and hope that **its proportion in the whole union region is as small as possible**, that is, the prediction tends to be consistent with the true value.

So, we can easily take the form of CEL, and the picture here shows the corresponding conception process of CEL.

![image](https://user-images.githubusercontent.com/26847524/104424782-c9a14700-55ba-11eb-8766-e13cabd41b7b.png)
==>>
![image](https://user-images.githubusercontent.com/26847524/104424921-f9504f00-55ba-11eb-81c5-97072b217c97.png)

In addition, **we analyze the effectiveness of cel, that is, the form of its gradient is more effective for this task than cross-entropy.**
> In Equ. (8), except that the numerator term 1 − 2g is position-specific,  **the other terms are image-specific**. And this numerator is closely related to the binary ground truth, **which results in that the inter-class derivatives have large differences while the intra-class ones are relatively consistent.**
> **This has several merits: 1) It ensures that there is enough large gradient to drive the network in the later stage of training; 2) It helps solve the intra-class inconsistency and inter-class indistinction issues, to some extent, thereby promoting the predicted boundaries of salient objects to become sharper.**
